extensions/amp-stream-gallery/0.1/amp-stream-gallery.js

```diff
@@ -0,0 +1,752 @@
+
+      setStyle(this.slidesContainer_, 'max-width', maxContainerWidth);
+    });
+    this.carousel_.updateSlides(this.slides_);
+    this.carousel_.updateAdvanceCount(advanceCount);
+    this.carousel_.updateSnapBy(advanceCount);
+    this.carousel_.updateVisibleCount(visibleCount);
+  }
+
+  /**
+   * @param {boolean} outsetArrows
+   * @private
+   */
+  updateOutsetArrows_(outsetArrows) {
+    this.outsetArrows_ = outsetArrows;
+    this.updateUi_();
+  }
+
+  /**
+   * @param {string} extraSpace
+   * @private
+   */
+  updateExtraSpace_(extraSpace) {
+    this.content_.setAttribute(
+      'i-amphtml-stream-gallery-extra-space',
+      extraSpace
+    );
+  }
+
+  /**
+   * @param {string} insetArrowVisibility
+   * @private
+   */
+  updateInsetArrowVisibility_(insetArrowVisibility) {
+    this.insetArrowVisibility_ =
+      insetArrowVisibility == 'always'
+        ? InsetArrowVisibility.ALWAYS
+        : insetArrowVisibility == 'never'
+        ? InsetArrowVisibility.NEVER
+        : InsetArrowVisibility.AUTO;
+    this.updateUi_();
+  }
+
+  /**
+   * Updates the UI of the <amp-base-carousel> itself, but not the internal
+   * implementation.
+   * @private
+   */
+  updateUi_() {
+    // TODO(sparhami) for Shadow DOM, we will need to get the assigned nodes
+    // instead.
+    iterateCursor(this.prevArrowSlot_.children, child => {
+      toggleAttribute(child, 'disabled', this.carousel_.isAtStart());
+    });
+    iterateCursor(this.nextArrowSlot_.children, child => {
+      toggleAttribute(child, 'disabled', this.carousel_.isAtEnd());
+    });
+    toggleAttribute(
+      dev().assertElement(this.content_),
+      'i-amphtml-stream-gallery-hide-inset-buttons',
+      this.shouldHideInsetButtons_()
+    );
+    toggleAttribute(
+      dev().assertElement(this.content_),
+      'amp-stream-gallery-outset-arrows',
+      this.outsetArrows_
+    );
+  }
+
+  /**
+   * Starts queuing all intersection based changes when scrolling starts, to
+   * prevent paint flickering on iOS.
+   */
+  onScrollStarted_() {
+    this.childLayoutManager_.setQueueChanges(this.isIos_);
+  }
+
+  /**
+   * Update the UI (buttons) for the new scroll position. This occurs when
+   * scrolling has settled.
+   */
+  onScrollPositionChanged_() {
+    // Now that scrolling has settled, flush any layout changes for iOS since
+    // it will not cause flickering.
+    this.childLayoutManager_.flushChanges();
+    this.childLayoutManager_.setQueueChanges(false);
+
+    this.updateUi_();
+  }
+
+  /**
+   * Updates the current index, triggering actions and analytics events.
+   * @param {number} index
+   * @param {!ActionSource} actionSource
+   */
+  updateCurrentIndex_(index, actionSource) {
+    const prevIndex = this.currentIndex_;
+    this.currentIndex_ = index;
+
+    // Ignore the first indexChange, we do not want to trigger any events.
+    if (prevIndex == null) {
+      return;
+    }
+
+    const data = dict({'index': index});
```

Comment:
Is this different from `dict({index})`/is the string key needed to prevent Closure from altering?

Reply:
Nvm; I see below that it probably is

Reply:
I think so but I'm not 100% sure.